docs/source/Loaders.rst

```diff
@@ -80,3 +80,119 @@ test categories. With tags set, users can select a subset of the
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_server_stderr
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_server_stdout
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_interactive_stdout
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_remote
+
+To filter the results, listing only the tests that have `test_interactive`
+in their test method names, you can execute::
+
+    $ avocado list /usr/share/avocado/tests/gdbtest.py:test_interactive
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_interactive
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_interactive_args
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_interactive_stdout
+
+As the string after the `:` is a regular expression, three tests were
+filtered in. You can manipulate the regular expression to have only the
+test with that exact name::
+
+    $ avocado list /usr/share/avocado/tests/gdbtest.py:test_interactive$
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_interactive
+
+The regular expression enables you to have more complex filters.
+Example::
+
+    $ avocado list /usr/share/avocado/tests/gdbtest.py:GdbTest.test_[le].*raw
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_existing_commands_raw
+    INSTRUMENTED /usr/share/avocado/tests/gdbtest.py:GdbTest.test_load_set_breakpoint_run_exit_raw
+
+Once the test reference is providing you the expected outcome, you can
+replace the `list` subcommand with the `run` subcommand to execute your
+tests::
+
+    $ avocado run /usr/share/avocado/tests/gdbtest.py:GdbTest.test_[le].*raw
+    JOB ID     : 333912fb02698ed5339a400b832795a80757b8af
+    JOB LOG    : $HOME/avocado/job-results/job-2017-06-14T14.54-333912f/job.log
+     (1/2) /usr/share/avocado/tests/gdbtest.py:GdbTest.test_existing_commands_raw: PASS (0.59 s)
+     (2/2) /usr/share/avocado/tests/gdbtest.py:GdbTest.test_load_set_breakpoint_run_exit_raw: PASS (0.42 s)
+    RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
+    JOB TIME   : 1.15 s
+    JOB HTML   : $HOME/avocado/job-results/job-2017-06-14T14.54-333912f/html/results.html
+
+External Loader
+---------------
+
+Using the External Loader, Avocado will consider that and External
+Runner will be in place and so Avocado doesn't really need to resolve
+the references. Instead, Avocado will pass the references as parameters
+to the External Runner. Example::
+
+    $ avocado run 20
+    Unable to resolve reference(s) '20' with plugins(s) 'file', 'robot',
+    'vt', 'external', try running 'avocado list -V 20' to see the details.
+
+In the command above, no loaders can resolve `20` as a test. But running
+the command above with the External Runner `/bin/sleep` will make Avocado
+to actually execute `/bin/sleep 20` and check for its return code::
+
+    $ avocado run 20 --loaders external:/bin/sleep
```

Comment:
especially with external runner it's very useful to put references at the end to avoid references becoming avocado arguments. Would you mind changing this example to `avocado run --loaders external:/bin/sleep -- 20`?